lib/blocs/home_page_cubit.dart

```diff
@@ -50,6 +51,16 @@ class HomePageCubit extends Cubit<MapRouteState> {
     emit(newState);
   }
 
+  Future<void> setTappingPlace(LatLng latLng) async {
```

Comment:
Just an unverified idea.
```dart
Future<void> setTappingPlace(LatLng latLng) async {
  await updateMapRouteState(
    state.copyWith(
        fromPlace: state.fromPlace ?? TrufiLocation.fromLatLng("Map Marker", latLng),
        toPlace: state.toPlace ?? TrufiLocation.fromLatLng("Map Marker", latLng)
    ),
  );
}
```

Reply:
when state.fromPlace == state.toPlace== null, the setTappingPlace defines the from an to with the same value.